### PR TITLE
feat: [CDS-38079]: OCI helm 3 support In NG

### PR DESCRIPTION
**Pull request by namantalaycha10**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33888)

### CODE REVIEW

**Comment by namantalaycha10** [comment]
trigger ti1

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 44 of 51 files at r1, 15 of 15 files at r2, 1 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33888)* status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java` line 532 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sDv36BW-_yCUA9xDR:-N3sDv36BW-_yCUA9xDS:bryoyqv) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java#L532)):*
> ```Java
>         if (!(connectorInfoDTO.getConnectorConfig() instanceof OciHelmConnectorDTO)) {
>           throw new InvalidRequestException(
>               format("Invalid connector selected in %s. Select Http Helm connector", message));
> ```

Should be select OCI Helm Repo connector
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java` line 682 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sDnUaC7Y9k_-vjZqN:-N3sDnUaC7Y9k_-vjZqO:ba06vxv) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java#L682)):*
> ```Java
>   }
> 
>   public boolean helmOciSupport(String accountId) {
> ```

name should be isHelmOciSupported/Enabled
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartStoreGenericConfig.java` line 41 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sFh3m3b91ObJYfft_:-N3sFh3n0QywvrfIurK5:b-ob9lo6) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartStoreGenericConfig.java#L41)):*
> ```Java
> @SimpleVisitorHelper(helperClass = ConnectorRefExtractorHelper.class)
> @TypeAlias("OciGenericConfig")
> @RecasterAlias("io/harness/cdng/manifest/yaml/OciHelmChartGenericConfig.java")
> ```

I don't think it should path but package + class name
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ociHelmChartConfig/OciHelmChartStoreConfigType.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sGKio5OiKewEDAFUh:-N3sGKio5OiKewEDAFUi:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ociHelmChartConfig/OciHelmChartStoreConfigType.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ociHelmChartConfig/OciHelmChartStoreConfigWrapper.java` line 53 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sGjcyEV-ysv8SUxG5:-N3sGjcyEV-ysv8SUxG6:b-rewq1r) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ociHelmChartConfig/OciHelmChartStoreConfigWrapper.java#L53)):*
> ```Java
>   private String uuid;
> 
>   @NotNull @JsonProperty("type") OciHelmChartStoreConfigType type;
> ```

nit: JsonProperty not required here, the name is already type
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ociHelmChartConfig/OciStoreConfigType.java` line 15 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sH83E-GZVNdksIBHC:-N3sH83E-GZVNdksIBHD:b-umly3x) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ociHelmChartConfig/OciStoreConfigType.java#L15)):*
> ```Java
> @OwnedBy(HarnessTeam.CDP)
> public interface OciStoreConfigType {
>   String GENERIC = "Generic";
> ```

We should use this in Enum
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ociHelmChartConfig/OciStoreConfigWrapperVisitorHelper.java` line 13 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sHCG00j6GqZjNTbVq:-N3sHCG00j6GqZjNTbVr:bapijl4) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ociHelmChartConfig/OciStoreConfigWrapperVisitorHelper.java#L13)):*
> ```Java
> import io.harness.walktree.visitor.validation.ValidationVisitor;
> 
> public class OciStoreConfigWrapperVisitorHelper implements ConfigValidator {
> ```

nit: missing ownedby
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/helm/OciHelmAuthentication.java` line 10 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sHcdz8AcQ4QS6bfaL:-N3sHcdz8AcQ4QS6bfaM:blfuw65) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/helm/OciHelmAuthentication.java#L10)):*
> ```Java
> package io.harness.connector.entities.embedded.helm;
> 
> public interface OciHelmAuthentication {}
> ```

missing ownedby
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/mappers/helm/OciHelmDTOToEntity.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sHnZq01ld2XLyn-em:-N3sHnZq01ld2XLyn-en:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/440-connector-nextgen/src/main/java/io/harness/connector/mappers/helm/OciHelmDTOToEntity.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/validator/OciHelmRepoConnectionValidator.java` line 48 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sWp-s5pxiRFQFugP-:-N3sWp-tBRQSp3qeCFk1:b-tg1ka5) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/440-connector-nextgen/src/main/java/io/harness/connector/validator/OciHelmRepoConnectionValidator.java#L48)):*
> ```Java
>   public ConnectorValidationResult validate(ConnectorConfigDTO connectorDTO, String accountIdentifier,
>       String orgIdentifier, String projectIdentifier, String identifier) {
>     OciHelmConnectivityTaskResponse responseData = (OciHelmConnectivityTaskResponse) super.validateConnector(
> ```

Shouldn't we create a custom validation for oci helm repo?
___
*[`810-ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/generator/OciHelmPollingItemGenerator.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sWxK5CkHzTg8khbeR:-N3sWxK5CkHzTg8khbeS:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/810-ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/generator/OciHelmPollingItemGenerator.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmConnectivityDelegateTask.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sXXgW6fgI5rytT8rG:-N3sXXgW6fgI5rytT8rH:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmConnectivityDelegateTask.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmConnectivityDelegateTask.java` line 61 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sXhwl4UPP7lpR4Tol:-N3sXhwl4UPP7lpR4Tom:b-qn9rch) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmConnectivityDelegateTask.java#L61)):*
> ```Java
>       throw new HelmClientRuntimeException(e);
>     } catch (Exception e) {
>       throw NestedExceptionUtils.hintWithExplanationException(e.getMessage(), "Failed to validate Oci Helm Repo", e);
> ```

nit: explanation and exception message will be the same, don't think is any utility here to throw hint and explantion
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmCapabilityHelper.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sY3Wa8p66Kispjez_:-N3sY3Wa8p66Kispjeza:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmCapabilityHelper.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmConnectivityTaskParams.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sY7LnCUu6F6ZSi6BT:-N3sY7LnCUu6F6ZSi6BU:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmConnectivityTaskParams.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmConnectivityTaskResponse.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sYACEE5XcOqrlVsMF:-N3sYACEE5XcOqrlVsMG:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmConnectivityTaskResponse.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmValidationParams.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sYEl45EGM9MJeII6H:-N3sYEl45EGM9MJeII6I:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmValidationParams.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/executioncapability/CapabilityType.java` line 53 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sYVs-DKO_37cK5dy_:-N3sYVs-DKO_37cK5dya:bs2z3x7) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/executioncapability/CapabilityType.java#L53)):*
> ```Java
>   ARTIFACTORY,
>   SERVERLESS_INSTALL,
>   OCI;
> ```

OCI_HELM_REPO?
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/storeconfig/OciHelmStoreDelegateConfig.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sYXrJ5TOFwpn3tWRx:-N3sYXrJ5TOFwpn3tWRy:bb2f3jn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/storeconfig/OciHelmStoreDelegateConfig.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

Should be 2022
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/storeconfig/OciHelmStoreDelegateConfig.java` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sYa-SAJ_veM_7V01N:-N3sYa-SAJ_veM_7V01O:b-ykrbsg) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/storeconfig/OciHelmStoreDelegateConfig.java#L25)):*
> ```Java
>   OciHelmConnectorDTO ociHelmConnector;
>   List<EncryptedDataDetail> encryptedDataDetails;
>   boolean helmOciSupport;
> ```

helmOciEnabled?
___
*[`954-connector-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmAuthenticationDTODeserializer.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sZ9k71A5ad_OgQmWV:-N3sZ9k71A5ad_OgQmWW:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmAuthenticationDTODeserializer.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022
___
*[`954-connector-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmAuthType.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sZBYF0EAlHrsYuPz2:-N3sZBYG0EaiYcMaCC4l:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmAuthType.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022
___
*[`954-connector-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmUsernamePasswordDTO.java` line 2 at r2](https://reviewable.io/reviews/harness/harness-core/33888#-N3sZJIY8v9MbwoSlgFU:-N3sZJIY8v9MbwoSlgFV:brmzafn) ([raw file](https://github.com/harness/harness-core/blob/844834f5b300cd7c86af722a47d714aa7f63c3f7/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmUsernamePasswordDTO.java#L2)):*
> ```Java
> /*
>  * Copyright 2021 Harness Inc. All rights reserved.
> ```

should be 2022

**Review by shubham-in10se — CHANGES_REQUESTED** [review]
> Reviewed 24 of 51 files at r1, 4 of 15 files at r2, 1 of 3 files at r3, 41 of 41 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33888)* status: 2 change requests. 20. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartConfig.java` line 55 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zOfBH9ffgofHqH6_6:-N3zOfBI1tG0mJSFIvPU:bxrpyu) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartConfig.java#L55)):*
> ```Java
> 
>   @EntityIdentifier String identifier;
>   @Wither
> ```

give a new line space here to better readability
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartConfig.java` line 59 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zOn0BCbLP5TeiIbqw:-N3zOn0BCbLP5TeiIbqx:bzaohgs) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartConfig.java#L59)):*
> ```Java
>   @ApiModelProperty(dataType = "io.harness.cdng.manifest.yaml.oci.OciHelmChartStoreConfigWrapper")
>   @SkipAutoEvaluation
>   ParameterField<OciHelmChartStoreConfigWrapper> config;
> ```

`identifier` and `config` fields are not marked `private`. Is this intentional .?
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartConfig.java` line 79 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zP3tCCONF0aZFE8Fx:-N3zP3tCCONF0aZFE8Fy:b-mrg04k) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/OciHelmChartConfig.java#L79)):*
> ```Java
> 
>   public StoreConfig applyOverrides(StoreConfig overrideConfig) {
>     OciHelmChartConfig ociHelmChartConfig = (OciHelmChartConfig) overrideConfig;
> ```

rename to `overrideHelmConfig`
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/helm/OciHelmConnector.java` line 33 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zS60L1JU2DoANgSYx:-N3zS60L1JU2DoANgSYy:bfcy9yz) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/helm/OciHelmConnector.java#L33)):*
> ```Java
> @TypeAlias("io.harness.connector.entities.embedded.helm.OciHelmConnector")
> public class OciHelmConnector extends Connector {
>   String url;
> ```

all fields are `package` privacy level. Is this intentional .? Why not have them `private`
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/helm/OciHelmUsernamePasswordAuthentication.java` line 25 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zSEsG3wzLqhjzUuqi:-N3zSEsG3wzLqhjzUuqj:bnj7hl0) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/helm/OciHelmUsernamePasswordAuthentication.java#L25)):*
> ```Java
>   String username;
>   String usernameRef;
>   String passwordRef;
> ```

same as above
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/heartbeat/OciHelmConnectorValidationParamsProvider.java` line 27 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zSHoH6Z_l17d6uY-X:-N3zSHoH6Z_l17d6uY-Y:byxy4jw) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/440-connector-nextgen/src/main/java/io/harness/connector/heartbeat/OciHelmConnectorValidationParamsProvider.java#L27)):*
> ```Java
> @Singleton
> public class OciHelmConnectorValidationParamsProvider implements ConnectorValidationParamsProvider {
>   @Inject EncryptionHelper encryptionHelper;
> ```

All dependencies should be marked `private`
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/validator/OciHelmRepoConnectionValidator.java` line 24 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zSiQu5K4fNlreoIkb:-N3zSiQu5K4fNlreoIkc:b-egdodl) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/440-connector-nextgen/src/main/java/io/harness/connector/validator/OciHelmRepoConnectionValidator.java#L24)):*
> ```Java
> import lombok.extern.slf4j.Slf4j;
> 
> @Slf4j
> ```

there is no `log` statement in the class. We can remove this.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmConnectivityDelegateTask.java` line 52 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zTwTt5M1t85vrbr69:-N3zTwTw9F3R6GCSxY5x:b8eorsp) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmConnectivityDelegateTask.java#L52)):*
> ```Java
>               .ociHelmConnectorDTO(ociHelmConnectorDTO)
>               .build();
>       ConnectorValidationResult ociHelmConnectorValidationResult =
> ```

can be renamed as `validationResult`, since we are getting all the other context from the rest of the statement/block.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java` line 69 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zUHLv2xV5-SviH1ES:-N3zUHLv2xV5-SviH1ET:b-ylxq63) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java#L69)):*
> ```Java
>       validationResultBuilder.status(ConnectivityStatus.SUCCESS);
>     } catch (IOException exception) {
>       log.warn("OciHelmValidationHandler execution failed with exception ", exception);
> ```

this should be an error log ..?
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmValidationParams.java` line 26 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zUZZi7MQLQUiLx-_c:-N3zUZZi7MQLQUiLx-_d:be4e4xa) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmValidationParams.java#L26)):*
> ```Java
> public class OciHelmValidationParams
>     extends ConnectorTaskParams implements ConnectorValidationParams, ExecutionCapabilityDemander {
>   OciHelmConnectorDTO ociHelmConnectorDTO;
> ```

why not make them all `private`
please also check the same for other defined data objects as well.
___
*[`954-connector-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmAuthenticationDTO.java` line 29 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zV1kz7d9n5g7dEn7Z:-N3zV1kz7d9n5g7dEn7_:bjuwolz) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/helm/OciHelmAuthenticationDTO.java#L29)):*
> ```Java
> @Schema(name = "OciHelmAuthentication", description = "This contains oci helm authentication details")
> public class OciHelmAuthenticationDTO {
>   @NotNull @JsonProperty("type") OciHelmAuthType authType;
> ```

`private` ..?
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciHelmChartStoreConfigType.java` line 21 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zQtDW8DpylNNPH9tJ:-N3zQtDW8DpylNNPH9tK:b-uwi514) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciHelmChartStoreConfigType.java#L21)):*
> ```Java
> public enum OciHelmChartStoreConfigType {
>   @JsonProperty(OciStoreConfigType.GENERIC) GENERIC(OciStoreConfigType.GENERIC);
>   private final String displayName;
> ```

add a new line space here for better readability
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciHelmChartStoreConfigType.java` line 39 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zR0LECPf6i4H5QyjO:-N3zR0LECPf6i4H5QyjP:bbi2mmw) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciHelmChartStoreConfigType.java#L39)):*
> ```Java
>   @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
>   public static OciHelmChartStoreConfigType getOciHelmChartStoreConfigType(@JsonProperty("type") String displayName) {
>     for (OciHelmChartStoreConfigType ociHelmChartStoreConfigType : OciHelmChartStoreConfigType.values()) {
> ```

always try to keep shorter variable names. Like here, we can simply use `configType`, and the rest of the context is coming from the Object Class name, viz. `OciHelmChartStoreConfigType`
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciHelmChartStoreConfigWrapper.java` line 57 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zRUdXE8HLiWJ9ZQY6:-N3zRUdXE8HLiWJ9ZQY7:b-why3t6) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciHelmChartStoreConfigWrapper.java#L57)):*
> ```Java
>   @JsonTypeInfo(use = NAME, property = "type", include = EXTERNAL_PROPERTY, visible = true)
>   @Wither
>   io.harness.cdng.manifest.yaml.oci.OciHelmChartStoreConfig spec;
> ```

why the need for a fully qualified name here ..?
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciHelmChartStoreConfigWrapper.java` line 99 at r4](https://reviewable.io/reviews/harness/harness-core/33888#-N3zRhMT0bfH5SN18PfF:-N3zRhMT0bfH5SN18PfG:b3vju8a) ([raw file](https://github.com/harness/harness-core/blob/d07764a4ec4531df736f816a865debb9193fb0a3/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciHelmChartStoreConfigWrapper.java#L99)):*
> ```Java
> 
>     public static OciHelmChartStoreConfigWrapperParameters fromOciHelmChartStoreConfigWrapper(
>         OciHelmChartStoreConfigWrapper ociHelmChartStoreConfigWrapper) {
> ```

same as above. The param name here can be `storeConfigWrapper`

**Review by namantalaycha10 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33888)* status: 2 change requests. 16. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciStoreConfigWrapperVisitorHelper.java` line 14 at r6](https://reviewable.io/reviews/harness/harness-core/33888#-N42zkSj1v2k188jeLjU:-N45lP3BCYaDnrZDTmtX:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/4b34e55c9d0220939e57f7adbec43ea244559548/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/oci/OciStoreConfigWrapperVisitorHelper.java#L14)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Should be right up to class
</blockquote></details>

Done.

**Comment by namantalaycha10** [comment]
Regarding the PR Codebasehash Check merge checklist

1. Yes new Task type is added in this
2. a Added one class is Kryo register, but since changes are behind a FF, we can wait to enable the feature for any customer until the changes go live in delegate release
3. yes changes are backward-compatible manner
4. yes Ng backward compatibility is there because is feature is for CG only
5. No removal of enum
6. All new enum are added in the end of the already defined constants

**Review by shubham-in10se — DISMISSED** [review]
> Reviewed 5 of 13 files at r5, 4 of 14 files at r6, 16 of 16 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33888)* status: 1 change requests. 1. unresolved discussions.

**Comment by namantalaycha10** [comment]
trigger ti0

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/HelmDeployServiceImplNG.java [hunk 1]
```diff
@@ -813,7 +818,8 @@ public List<KubernetesResource> printHelmChartKubernetesResources(HelmInstallCom
             .filter(Objects::nonNull)
             .filter(storeType
                 -> storeType == StoreDelegateConfigType.S3_HELM || storeType == StoreDelegateConfigType.HTTP_HELM
-                    || storeType == StoreDelegateConfigType.GIT || storeType == StoreDelegateConfigType.GCS_HELM);
+                    || storeType == StoreDelegateConfigType.GIT || storeType == StoreDelegateConfigType.GCS_HELM
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/HelmDeployServiceImplNG.java:821]: Create a static list with all the types and do a contains check. This is not readable now.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by mdmilic — APPROVED** [review]
Approve delegate related changes and codebase hash update

> Reviewed 7 of 51 files at r1, 7 of 41 files at r4, 1 of 16 files at r7, 2 of 4 files at r9, 2 of 3 files at r10.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33888#-:-N47UYTi4_uxNLFtwNc-:b-wq7n8p)* status: 1 file(s) to review. 2 change requests. 1. unresolved discussions.

**Pull request merged**